### ISSUE TITLE
Search term lost when changing date

### ISSUE THREAD

**Issue by goochj03** [piwik/piwik#21551]
When performing a search for a URL or page title on a report e.g. Behaviour > Pages, changing the date seems to clear the text that was entered into the search field. Having the text remain in the search field, would provide a more seamless user experience.

One customer example:
"When I look at e.g. a page report and do a keyword search for a URL or a page title, then it would be helpful, if I could change the date range and not loose the current search term."

**Comment by michalkleiner** [comment]
Thanks for raising the issue @goochj03.
I think this is a question for product as changing the date changes it for all the widgets and reports in the view whereas the search is within the single widget, so it's a UX/product decision what selections and filters should be sticky and persist across e.g. date changes, site changes and similar.

**Comment by Stan-vw** [comment]
To check if I understand the situation + requested functionality here:
- you're currently using the search functionality as a sort of filter on the data
- you'd like that filter to stay active when changing other filters (like date/segment/...)

I think we should consider if we should do it through
- "keeping the search term", or 
- some functionality to turn the search term into an actual filter. That way, you might be able to stack multiple search terms

**Comment by goochj03** [comment]
Yes this is when using the 'Search' option located in the bottom left of each report. e.g. Behaviour > Pages
![image](https://github.com/matomo-org/matomo/assets/118518962/4d374c01-b7c7-493f-ba35-d5dc6903f61d)

Entering a search word, then changing the date via the calendar resets the search field and displays the default report.

**Comment by goochj03** [comment]
Another customer has mentioned the search filter being reset when the date is adjusted "If we use a filter in search and change the date after, the filter is dismissed and we have to put it again."

**Comment by Chardonneaur** [comment]
I got a client asking for this feature today. In fact when a filter is active, it should stay active.

**Comment by Chardonneaur** [comment]
Another client asked for this feature today.
